src/avif.c

```diff
@@ -721,6 +721,54 @@ avifBool avifCleanApertureBoxConvertCropRect(avifCleanApertureBox * clap,
     return AVIF_TRUE;
 }
 
+// ---------------------------------------------------------------------------
+
+avifBool avifAreGridDimensionsValid(avifPixelFormat yuvFormat, uint32_t columns, uint32_t rows, uint32_t tileW, uint32_t tileH, avifDiagnostics * diag)
+{
+    // The rules below only apply to grid images. Single-cell images do not have these constraints.
+    if ((columns == 1) && (rows == 1)) {
+        return AVIF_TRUE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - the tile_width shall be greater than or equal to 64, and should be a multiple of 64
+    //   - the tile_height shall be greater than or equal to 64, and should be a multiple of 64
+    // The "should" part is ignored here.
+    if ((tileW < 64) || (tileH < 64)) {
+        avifDiagnosticsPrintf(diag,
+                              "[Strict] Grid image tile width (%u) or height (%u) cannot be smaller "
+                              "than 64 if there is more than one cell. "
+                              "See MIAF (ISO/IEC 23000-22:2019), Section 7.3.11.4.2",
+                              tileW,
+                              tileH);
+        return AVIF_FALSE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - when the images are in the 4:2:2 chroma sampling format the horizontal tile offsets and widths,
+    //     and the output width, shall be even numbers;
+    //   - when the images are in the 4:2:0 chroma sampling format both the horizontal and vertical tile
+    //     offsets and widths, and the output width and height, shall be even numbers.
+    // Only the tile dimensions are enforced since tile offsets and image dimensions depend on them.
+    // If the rules above were not respected, the following problematic situation may happen:
+    //   Some 4:2:0 image is Pi pixels wide and has N cell columns.
+    //   Each cell is Pc pixels wide, Pc being odd.
```

Comment:
I believe you also assume `Pi = Pc*N`, right?

Note that in general this assumption is not true. Please see Figure 2 on page 18 of MIAF. It shows that in general grid width doesn't need to be equal to output width, and grid height doesn't need to be equal to output height.